src/test/java/com/wikia/webdriver/common/core/drivers/BrowserType.java

```diff
@@ -0,0 +1,59 @@
+package com.wikia.webdriver.common.core.drivers;
+
+import com.wikia.webdriver.common.core.WikiaWebDriver;
+import com.wikia.webdriver.common.core.drivers.browsers.AndroidBrowser;
+import com.wikia.webdriver.common.core.drivers.browsers.ChromeBrowser;
+import com.wikia.webdriver.common.core.drivers.browsers.ChromeMobileBrowser;
+import com.wikia.webdriver.common.core.drivers.browsers.DefaultBrowser;
+import com.wikia.webdriver.common.core.drivers.browsers.FirefoxBrowser;
+import com.wikia.webdriver.common.core.drivers.browsers.GhostBrowser;
+import com.wikia.webdriver.common.core.drivers.browsers.HtmlUnitBrowser;
+
+public enum BrowserType{
+  CHROME(ChromeBrowser.class, "CHROME"),
+  FIREFOX(FirefoxBrowser.class, "FF"),
+  CHROME_MOBILE(ChromeMobileBrowser.class, "CHROMEMOBILEMERCURY"),
+  HTMLUNIT(HtmlUnitBrowser.class, "HTMLUNIT"),
+  GHOST(GhostBrowser.class, "GHOST"),
+  CHROME_ANDROID(AndroidBrowser.class, "ANDROID"),
+  DEFAULT(DefaultBrowser.class, "");
+
+  private Class<? extends BrowserAbstract> browser;
+  private String name;
+
+  BrowserType(Class<? extends BrowserAbstract> browser, String name) {
+    this.name = name;
+    this.browser = browser;
+  }
+
+  public String getName(){
+    return name;
+  }
+
+  public WikiaWebDriver getInstance() {
+    try {
+      return browser.newInstance().getInstance();
+    } catch (InstantiationException | IllegalAccessException e) {
+      e.printStackTrace();
+    }
+    return null;
+  }
+
+  public WikiaWebDriver setInstance() {
+    try {
+      return browser.newInstance().setInstance();
+    } catch (InstantiationException | IllegalAccessException e) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Either log or rethrow this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar-s1/coding_rules#rule_key=squid%3AS1166)